cmd/bucket-handlers.go

```diff
@@ -606,23 +616,18 @@ func (api objectAPIHandlers) DeleteMultipleObjectsHandler(w http.ResponseWriter,
 			}
 		}
 
-		if hasLifecycleConfig && dobj.PurgeTransitioned == lifecycle.TransitionComplete { // clean up transitioned tier
-			delTier := false
-			switch {
-			case dobj.VersionID != "":
-				delTier = true
-			case !globalBucketVersioningSys.Enabled(bucket):
-				delTier = true
-			}
-			if delTier {
-				deleteTransitionedObject(ctx, newObjectLayerFn(), bucket, dobj.ObjectName, lifecycle.ObjectOpts{
-					Name:         dobj.ObjectName,
-					VersionID:    dobj.VersionID,
-					DeleteMarker: dobj.DeleteMarker,
-				}, dobj.TransitionedObjName, dobj.TransitionTier, false, false)
+	}
+
+	// Clean up transitioned objects from remote tier
+	if hasLifecycleConfig {
+		for _, os := range oss {
```

Comment:
If there are errors due to permissions or object locking constraints, the oss can have null entries - need to guard against panics